TypeCobol.LanguageServer/Completion Factory/CompletionFactory.cs

```diff
@@ -229,8 +229,24 @@ public static IEnumerable<CompletionItem> GetCompletionForProcedureParameter(Pos
                 var parameterToFill = procParams.ToArray()[alreadyGivenParametersCount];
                 //Get local/global variable that could correspond to the parameter
 
-                potentialVariablesForCompletion = node.SymbolTable.GetVariablesByType(parameterToFill.DataType, potentialVariablesForCompletion, SymbolTable.Scope.Program);
+                DataType paramDataType = parameterToFill.DataType;
+                string paramQualifiedNameTail = string.Empty;
+                if (paramDataType.Name.Contains("."))
+                {
+                    // parameter has a qualified name 
+                    int pos = paramDataType.Name.IndexOf(".");
+                    paramQualifiedNameTail = paramDataType.Name.Substring(0, pos);
+                    paramDataType = new DataType(paramDataType.Name.Substring(pos + 1), paramDataType.RestrictionLevel, paramDataType.CobolLanguageLevel);
+                }
+                List<DataDefinition> variablesForCompletion = node.SymbolTable.GetVariablesByType(paramDataType, potentialVariablesForCompletion, SymbolTable.Scope.Program);
```

Comment:
It works but, this code still rely on DataType.
The DataType the exact text written in the TypeCobol source to references a Type.
The property TypeDefinition is a link to a TypeDefinition node if the node is typed.
Property TypeDefinition is more accurate than DataType because you are sure that the link is correct.

So method SymbolTable.GetVariablesByType should be rewritten to use a TypeDefinition instead of a DataType.

Reply:
done
https://github.com/TypeCobolTeam/TypeCobol/pull/1683/commits/ea9a7fea4aeb71af6fb64cc6e4b4a87a5e9108c8